rosette/rbl/rosette/produce.rbl

```diff
@@ -7,14 +7,14 @@
 (defOprn fetch-pattern-and-product)
 (defProc (fetch-pattern-and-product chart channel product-patterns channel-index channel-position)
   (if (= channel-index channel-position)
-    [channel -1]
-    (let [[[data-list continuation-struct-list] (tbl-get chart channel)]]
-      (if (null? data-list)
-        []
-        (iterate loop [[[data & r] data-list] [data-index 0]]
-          (if (match-pattern (tuple-safe-nth product-patterns channel-index) data)
-            [channel data-index]
-            (loop r (inc data-index))))))))
+      [channel -1]
+      (let [[[data-list continuation-struct-list] (tbl-get chart channel)]]
+        (if (null? data-list)
+            []
+            (iterate loop [[[data & r] data-list] [data-index 0]]
+                     (if (match-pattern (tuple-safe-nth product-patterns channel-index) data)
```

Comment:
What is the reason for the (if ...) being indented until the loop but for things like 

```
           (proc [candidate]
             (if (null? candidate)
```

the if is only indented by two spaces?